Source/WTF/wtf/ThreadSafeWeakPtr.h

```diff
@@ -136,6 +143,8 @@ class ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr {
 public:
     void ref() const { m_controlBlock.strongRef(); }
     void deref() const { m_controlBlock.template strongDeref<destructionThread>(); }
+    bool hasOneStrongRef() const { return m_controlBlock.strongRefCount() == 1;  }
```

Comment:
I seem to remember @achristensen07 purposefully didn’t add this one because it is racy?

Reply:
See my [comment](https://github.com/WebKit/WebKit/pull/8915/files#r1083243910) in `ImageBuffer.cpp` below.

The `.hasOneRef()` was racy before, so there's technically no behavior change, but this is less-than-ideal.

Reply:
Woke up early this morning and a possible solution popped into my head.

Could we create a template function like this:

```
template <typename T, typename U>
U ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<T>::runWithRefCount(lambda_expr(size_t refCount) -> U);
```

This method would take the lock, then run _lambda_expr_ with the current ref count.

For the case in `ImageBuffer.cpp`, though, I'm not sure if this is really protecting anything.

There are a couple more instances where this is needed in debug assertions that are just `ASSERT(hasOneRef())`.  Maybe we need to figure out why `hasOneRef()` is a special state for this code.  Do you know @shallawa?